powerhome/playbook
# Automated changelog

#### Description

Adds the ability to configure a brand new shiny changelog generator. 
The changelog will now be run as a command when the person releasing for the week does the release steps.

#### Breaking Changes

None

#### Runway Ticket URL

https://nitro.powerhrg.com/runway/backlog_items/NUX-808

#### How to test this
Docs for the library are here: https://github.com/github-changelog-generator/github-changelog-generator

These commands are still up for discussion as far as how to run them and the exact flags still need fleshing out before we solidify them for documentation: 
`bin/bundle exec github_changelog_generator` For now.. that will get you running. Alternatively you can add the `--since-tag 6.0.1` to limit your changes. (This is now added in the config file of Playbook, but will need to be added to our script to be updated automatically) Finally, you can add a github token and flag it with `--token XXXXXXXXXyour-40-character-token-hereXXXXXXXXXXX` to get past any rate limit issues. [Click me to be redirected and generate said token please.](https://github.com/settings/tokens/new?description=GitHub%20Changelog%20Generator%20token)

Advanced examples are here: 
https://github.com/github-changelog-generator/github-changelog-generator/wiki/Advanced-change-log-generation-examples  

We are working through those to see what flags we will need and how we want to use our labels in conjunction with them. In particular the `add-sections` flag seems promising. All of these are configured in our `github_generator_changelog` file at the root of the repo.

#### Checklist:
